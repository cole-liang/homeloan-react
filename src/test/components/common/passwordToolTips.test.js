import React from "react";
import { shallow } from "enzyme";
import PasswordToolTips from "../../../components/common/passwordTooltips";

describe("PasswordToolTips Component Test", () => {
  const pwdCharactersNumError = "Password must have 6-16 characters";
  const pwdAlphaDigitError = "At least one alphabet and one number";
  const pwdTypeError = "Contains only 0-9, a-z, A-Z and _";
  const possibleErrors = {
    pwdCharactersNumError,
    pwdAlphaDigitError,
    pwdTypeError
  };

  it("should render PasswordToolTips without errors", () => {
    const props = { possibleErrors, passwordErrors: [] };
    const wrapper = shallow(<PasswordToolTips {...props} />).dive();
    expect(wrapper).toMatchSnapshot();

    expect(
      wrapper
        .find("StyledComponent")
        .at(0)
        .prop("isPwdAlphaDigitInvalid")
    ).toBe(false);
    expect(
      wrapper
        .find("StyledComponent")
        .at(0)
        .prop("isPwdCharactersNumInvalid")
    ).toBe(false);
    expect(
      wrapper
        .find("StyledComponent")
        .at(0)
        .prop("isPwdTypeInvalid")
    ).toBe(false);
  });

  it("should render PasswordToolTips with all errors", () => {
    const props = {
      possibleErrors,
      passwordErrors: [pwdCharactersNumError, pwdAlphaDigitError, pwdTypeError]
    };
    const wrapper = shallow(<PasswordToolTips {...props} />).dive();
    expect(wrapper).toMatchSnapshot();
    expect(
      wrapper
        .find("StyledComponent")
        .at(0)
        .prop("isPwdAlphaDigitInvalid")
    ).toBe(true);
    expect(
      wrapper
        .find("StyledComponent")
        .at(0)
        .prop("isPwdCharactersNumInvalid")
    ).toBe(true);
    expect(
      wrapper
        .find("StyledComponent")
        .at(0)
        .prop("isPwdTypeInvalid")
    ).toBe(true);
  });
});
