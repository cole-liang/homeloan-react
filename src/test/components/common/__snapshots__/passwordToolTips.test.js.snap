// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordToolTips Component Test should render PasswordToolTips with all errors 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  & .pwdToolTipRow {
    margin: 0px 0px 10px 0px;
    opacity: 1 !important;
  }

  & .pwdToolTipRow:last-child {
    margin: 0px;
  }

  & #pwdCharactersNumErrorText {
    color: ",
          [Function],
          ";
  }

  & #pwdCharactersNumErrorIcon {
    margin-left: 30px;
  }

  & #pwdCharactersNumErrorIcon .fa-check-circle {
    display: ",
          [Function],
          ";
  }

  & #pwdCharactersNumErrorIcon .fa-times-circle {
    display: ",
          [Function],
          ";
  }

  & #pwdAlphaDigitErrorText {
    color: ",
          [Function],
          ";
  }

  & #pwdAlphaDigitErrorIcon {
    margin-left: 16px;
  }

  & #pwdAlphaDigitErrorIcon .fa-check-circle {
    display: ",
          [Function],
          ";
  }

  & #pwdAlphaDigitErrorIcon .fa-times-circle {
    display: ",
          [Function],
          ";
  }

  & #pwdTypeErrorText {
    color: ",
          [Function],
          ";
  }

  & #pwdTypeErrorIcon {
    margin-left: 53px;
  }

  & #pwdTypeErrorIcon .fa-check-circle {
    display: ",
          [Function],
          ";
  }

  & #pwdTypeErrorIcon .fa-times-circle {
    display: ",
          [Function],
          ";
  }

  & i {
    font-size: 17px;
  }

  & i.fa-check-circle {
    color: green;
  }

  & i.fa-times-circle {
    color: red;
  }
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  isPwdAlphaDigitInvalid={true}
  isPwdCharactersNumInvalid={true}
  isPwdTypeInvalid={true}
>
  <Bootstrap(Row)
    className="pwdToolTipRow justify-content-between"
    key="pwdCharactersNumErrorRow"
  >
    <span
      id="pwdCharactersNumErrorText"
    >
      Password must have 6-16 characters
    </span>
    <span
      id="pwdCharactersNumErrorIcon"
    >
      <i
        className="fa fa-check-circle"
      />
      <i
        className="fa fa-times-circle"
      />
    </span>
  </Bootstrap(Row)>
  <Bootstrap(Row)
    className="pwdToolTipRow justify-content-between"
    key="pwdAlphaDigitErrorRow"
  >
    <span
      id="pwdAlphaDigitErrorText"
    >
      At least one alphabet and one number
    </span>
    <span
      id="pwdAlphaDigitErrorIcon"
    >
      <i
        className="fa fa-check-circle"
      />
      <i
        className="fa fa-times-circle"
      />
    </span>
  </Bootstrap(Row)>
  <Bootstrap(Row)
    className="pwdToolTipRow justify-content-between"
    key="pwdTypeErrorRow"
  >
    <span
      id="pwdTypeErrorText"
    >
      Contains only 0-9, a-z, A-Z and _
    </span>
    <span
      id="pwdTypeErrorIcon"
    >
      <i
        className="fa fa-check-circle"
      />
      <i
        className="fa fa-times-circle"
      />
    </span>
  </Bootstrap(Row)>
</StyledComponent>
`;

exports[`PasswordToolTips Component Test should render PasswordToolTips without errors 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  & .pwdToolTipRow {
    margin: 0px 0px 10px 0px;
    opacity: 1 !important;
  }

  & .pwdToolTipRow:last-child {
    margin: 0px;
  }

  & #pwdCharactersNumErrorText {
    color: ",
          [Function],
          ";
  }

  & #pwdCharactersNumErrorIcon {
    margin-left: 30px;
  }

  & #pwdCharactersNumErrorIcon .fa-check-circle {
    display: ",
          [Function],
          ";
  }

  & #pwdCharactersNumErrorIcon .fa-times-circle {
    display: ",
          [Function],
          ";
  }

  & #pwdAlphaDigitErrorText {
    color: ",
          [Function],
          ";
  }

  & #pwdAlphaDigitErrorIcon {
    margin-left: 16px;
  }

  & #pwdAlphaDigitErrorIcon .fa-check-circle {
    display: ",
          [Function],
          ";
  }

  & #pwdAlphaDigitErrorIcon .fa-times-circle {
    display: ",
          [Function],
          ";
  }

  & #pwdTypeErrorText {
    color: ",
          [Function],
          ";
  }

  & #pwdTypeErrorIcon {
    margin-left: 53px;
  }

  & #pwdTypeErrorIcon .fa-check-circle {
    display: ",
          [Function],
          ";
  }

  & #pwdTypeErrorIcon .fa-times-circle {
    display: ",
          [Function],
          ";
  }

  & i {
    font-size: 17px;
  }

  & i.fa-check-circle {
    color: green;
  }

  & i.fa-times-circle {
    color: red;
  }
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  isPwdAlphaDigitInvalid={false}
  isPwdCharactersNumInvalid={false}
  isPwdTypeInvalid={false}
>
  <Bootstrap(Row)
    className="pwdToolTipRow justify-content-between"
    key="pwdCharactersNumErrorRow"
  >
    <span
      id="pwdCharactersNumErrorText"
    >
      Password must have 6-16 characters
    </span>
    <span
      id="pwdCharactersNumErrorIcon"
    >
      <i
        className="fa fa-check-circle"
      />
      <i
        className="fa fa-times-circle"
      />
    </span>
  </Bootstrap(Row)>
  <Bootstrap(Row)
    className="pwdToolTipRow justify-content-between"
    key="pwdAlphaDigitErrorRow"
  >
    <span
      id="pwdAlphaDigitErrorText"
    >
      At least one alphabet and one number
    </span>
    <span
      id="pwdAlphaDigitErrorIcon"
    >
      <i
        className="fa fa-check-circle"
      />
      <i
        className="fa fa-times-circle"
      />
    </span>
  </Bootstrap(Row)>
  <Bootstrap(Row)
    className="pwdToolTipRow justify-content-between"
    key="pwdTypeErrorRow"
  >
    <span
      id="pwdTypeErrorText"
    >
      Contains only 0-9, a-z, A-Z and _
    </span>
    <span
      id="pwdTypeErrorIcon"
    >
      <i
        className="fa fa-check-circle"
      />
      <i
        className="fa fa-times-circle"
      />
    </span>
  </Bootstrap(Row)>
</StyledComponent>
`;
